Panic when map lamba function produces an error

```
url = "https://hooks.slack.com"
from(bucket: "_monitoring")
    |> range(start: -1d)
    |> filter(fn: (r) => r._measurement == "statuses")
    |> last()  
    |> map(fn: (r) =>  ({
        status: slack.message(
            url: url,
            channel: "#test",    
            text: "The last reported status was \"${r.nonexistantKey}\".",
            color: "good"
     )}))
```

The data does not have the column `nonexistantKey` and so produces an error when interpolating the string. That error manifests as a panic from inside map about an unexpected kind. 

Example panic:

```
panic: unexpected kind: got "invalid" expected "string", trace: goroutine 184199 [running]:
 runtime/debug.Stack(0x14, 0xc008c88c80, 0xc009b79190)
 /usr/local/go/src/runtime/debug/stack.go:24 +0x9f github.com/influxdata/flux/values.UnexpectedKind(0x0, 0x1, 0x0, 0x0)
 /idpe/vendor/github.com/influxdata/flux/values/values.go:287 +0x36 github.com/influxdata/flux/values.null.Str(...)
 /idpe/vendor/github.com/influxdata/flux/values/values.go:306 github.com/influxdata/flux/compiler.(*stringExpressionEvaluator).Eval(0xc00c188f00, 0x3e637e0, 0xc007ee4d20, 0x3e8d520, 0xc000f5c710, 0x3e91640, 0xc00950e330, 0x0, 0x0)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:153 +0xf3 github.com/influxdata/flux/compiler.eval(...)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:86 github.com/influxdata/flux/compiler.(*objEvaluator).Eval.func1(0xc00c192b60, 0xc00c192b60, 0xc00fa62700)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:205 +0x15e github.com/influxdata/flux/values.BuildObjectWithSize(0x0, 0xc009b794f8, 0x3b53c40, 0x8, 0x3e220e0, 0x7fb987a1c7e0)
 /idpe/vendor/github.com/influxdata/flux/values/object.go:114 +0xde github.com/influxdata/flux/values.BuildObject(...)
 /idpe/vendor/github.com/influxdata/flux/values/object.go:101 github.com/influxdata/flux/compiler.(*objEvaluator).Eval(0xc009653bc0, 0x3e637e0, 0xc007ee4d20, 0x3e8d520, 0xc000f5c710, 0xc00999b568, 0xc009653aa0, 0xc00999b5e8, 0xc00999b560)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:193 +0x99 github.com/influxdata/flux/compiler.(*callEvaluator).Eval(0xc011098140, 0x3e637e0, 0xc007ee4d20, 0x3e8d520, 0xc000f5c710, 0x40eed0, 0xc00c192b00, 0x20, 0x18)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:539 +0x6e github.com/influxdata/flux/compiler.eval(...)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:86 github.com/influxdata/flux/compiler.(*objEvaluator).Eval.func1(0xc00c192b00, 0xc00c192b00, 0xc00999b680)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:205 +0x15e github.com/influxdata/flux/values.BuildObjectWithSize(0x0, 0xc009b797d0, 0x41048f, 0x3cddd60, 0xc00999b680, 0xc009b797f0)
 /idpe/vendor/github.com/influxdata/flux/values/object.go:114 +0xde github.com/influxdata/flux/values.BuildObject(...)
 /idpe/vendor/github.com/influxdata/flux/values/object.go:101 github.com/influxdata/flux/compiler.(*objEvaluator).Eval(0xc009653dd0, 0x3e637e0, 0xc007ee4d20, 0x3e8d520, 0xc000f5c710, 0xc000f5c710, 0x551b350, 0x1, 0x7fb98fbca3f0)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:193 +0x99 github.com/influxdata/flux/compiler.eval(...)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:86 github.com/influxdata/flux/compiler.(*blockEvaluator).Eval(0xc011098180, 0x3e637e0, 0xc007ee4d20, 0x3e8d520, 0xc000f5c710, 0xc008c88cd0, 0x0, 0x0, 0x92781d)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:106 +0x9a github.com/influxdata/flux/compiler.eval(...)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:86 github.com/influxdata/flux/compiler.compiledFn.Eval(0x3e21760, 0xc011098180, 0x3e8d520, 0xc000f5c710, 0x3e637e0, 0xc007ee4d20, 0x3e94d80, 0xc008c88cd0, 0x3e220e0, 0xc007dce4f0, ...)
 /idpe/vendor/github.com/influxdata/flux/compiler/runtime.go:50 +0xbb github.com/influxdata/flux/execute.(*rowFn).eval(0xc011098200, 0x3e637e0, 0xc007ee4d20, 0x0, 0x3e8bd20, 0xc008c88c38, 0x0, 0xc011098200, 0x8aaf09, 0xc00a6eba00, ...)
 /idpe/vendor/github.com/influxdata/flux/execute/row_fn.go:195 +0x21f github.com/influxdata/flux/execute.(*RowMapPreparedFn).Eval(0xc011098200, 0x3e637e0, 0xc007ee4d20, 0x0, 0x3e8bd20, 0xc008c88c38, 0xc000680800, 0x0, 0x8, 0x0)
 /idpe/vendor/github.com/influxdata/flux/execute/row_fn.go:285 +0x77 github.com/influxdata/flux/stdlib/universe.(*mapTransformation).Process.func1(0x3e8bd20, 0xc008c88c38, 0x3e2ca60, 0xc00c192800)
 /idpe/vendor/github.com/influxdata/flux/stdlib/universe/map.go:155 +0x105 github.com/influxdata/idpe/influxdbv2/storage/flux.(*windowTableRow).Do(0xc008c88c30, 0xc009653e60, 0x0, 0x0)
 /idpe/influxdbv2/storage/flux/window.go:150 +0x90 github.com/influxdata/flux/stdlib/universe.(*mapTransformation).Process(0xc00a11b1d0, 0x1d5bc0014adf474c, 0x94b904a120b6369b, 0x3e6e2e0, 0xc008c88c30, 0x4, 0x0)
 /idpe/vendor/github.com/influxdata/flux/stdlib/universe/map.go:152 +0xf5 github.com/influxdata/flux/execute.processMessage(0x3e637e0, 0xc007ee4d20, 0x3e7f3c0, 0xc00a11b1d0, 0x3e21da0, 0xc008c8d720, 0xc009e6d6e0, 0xc009e6d6e0, 0xc00540b740)
 /idpe/vendor/github.com/influxdata/flux/execute/transport.go:243 +0x312 github.com/influxdata/flux/execute.(*consecutiveTransport).processMessages(0xc002517ea0, 0x3e637e0, 0xc007ee4d20, 0xa)
 /idpe/vendor/github.com/influxdata/flux/execute/transport.go:191 +0xba github.com/influxdata/flux/execute.(*poolDispatcher).run(0xc004be4420, 0x3e637e0, 0xc007ee4d20)
 /idpe/vendor/github.com/influxdata/flux/execute/dispatcher.go:132 +0x67 github.com/influxdata/flux/execute.(*poolDispatcher).Start.func1(0xc004be4420, 0x3e637e0, 0xc007ee4d20)
 /idpe/vendor/github.com/influxdata/flux/execute/dispatcher.go:86 +0x89 created by github.com/influxdata/flux/execute.(*poolDispatcher).Start /idpe/vendor/github.com/influxdata/flux/execute/dispatcher.go:63 +0x85 
```

--- a/compiler/compiler_test.go
+++ b/compiler/compiler_test.go
@@ -55,6 +55,19 @@ func TestCompileAndEval(t *testing.T) {
 			wantCompileErr: true,
 		},
 		{
+			name: "interpolated string expression null",
+			fn:   `(r) => "n = ${r.n}"`,
+			inType: semantic.NewObjectType([]semantic.PropertyType{
+				{Key: []byte("r"), Value: semantic.NewObjectType([]semantic.PropertyType{
+					{Key: []byte("n"), Value: semantic.BasicString},
+				})},
+			}),
+			input: values.NewObjectWithValues(map[string]values.Value{
+				"r": values.NewObjectWithValues(map[string]values.Value{}),
+			}),
+			wantEvalErr: true,
+		},
+		{
 			name: "simple ident return",
 			fn:   `(r) => r`,
 			inType: semantic.NewObjectType([]semantic.PropertyType{

--- a/compiler/runtime.go
+++ b/compiler/runtime.go
@@ -150,6 +150,10 @@ func (e *stringExpressionEvaluator) Eval(ctx context.Context, scope Scope) (valu
 		if err != nil {
 			return nil, err
 		}
+
+		if v.IsNull() {
+			return nil, errors.New(codes.Invalid, "string expression evaluated to null")
+		}
 		b.WriteString(v.Str())
 	}
 	return values.NewString(b.String()), nil

